src/freehand_selection.jl

```diff
@@ -1,61 +1,188 @@
 function init_freehand_select(ctx::ImageContext)
     c = ctx.canvas
     enabled = Signal(true)
-    dragging = Signal(false)
+    drawing = Signal(false)
+    moving = Signal(false)
+    diff = Signal(XY(NaN,NaN))
 
     dummybtn = MouseButton{UserUnit}()
     
     sigstart = map(filterwhen(enabled, dummybtn, c.mouse.buttonpress)) do btn
         if !ispolygon(value(ctx.points)) || !isinside(Point(btn.position), Point.(value(ctx.points))) # prevents conflict with init_move_polygon
-            push!(dragging, true)
+            push!(drawing, true)
             push!(ctx.shape, Rectangle()) # some identifier of type of selection
             push!(ctx.points, [])
```

Comment:
Not sure about the syntax of typing an empty vector. Would it be "Vector{XY}[]"?